### PR TITLE
Add benchmark tests for pallet bags list

### PR DESCRIPTION
**Pull request by emostov**
Benchmark tests were accidentally left out of the bags list pallet; this PR adds them.

Additionally this PR fixes a regression to the staking pallet's implementation of `set_vote_weight_of`, a benchmark helper function. (See https://github.com/paritytech/substrate/pull/10593#discussion_r779131001)

### CODE REVIEW

**Comment by shawntabrizi** [comment]
@ggwpez mentioned that some of the tests failed when adding this. Did you try running tests locally?

**Comment by emostov** [comment]
> @ggwpez mentioned that some of the tests failed when adding this. Did you try running tests locally?

Tests are working locally for me (the mock runtime just needs to be setup slightly different then the default; see `skip_genesis_ids()`) - but currently working on trying to figure out why `cargo run --features=runtime-benchmarks --manifest-path=bin/node/cli/Cargo.toml -- benchmark --chain=dev --steps=1 --repeat=1 --pallet='pallet-bags-list' --extrinsic='*' --execution=wasm --wasm-execution compiled` is not working

File: frame/staking/src/pallet/impls.rs [hunk 1]
```diff
@@ -1245,13 +1245,12 @@ impl<T: Config> VoteWeightProvider<T::AccountId> for Pallet<T> {
 		// this will clearly results in an inconsistent state, but it should not matter for a
 		// benchmark.
 		let active: BalanceOf<T> = weight.try_into().map_err(|_| ()).unwrap();
-		let ledger = match Self::ledger(who) {
+		let mut ledger = match Self::ledger(who) {
 			None => StakingLedger::default_from(who.clone()),
```
Comment (emostov) [inline frame/staking/src/pallet/impls.rs:1249]: We expect benchmarks for the bags list to access the case where the ledger does not already exist (assuming the list is configured to use the staking pallet as the vote weight provider). AFAICT it was just an oversight that this did not set the `active` in the case we had to create a new ledger, but I want to make sure I understand why the benchmarks where working before in polkadot

Comment (emostov) [reply frame/staking/src/pallet/impls.rs:1249]: Found the regression: https://github.com/paritytech/substrate/commit/55cd07a7e22c26932f7bd16b87ea5a7569e38eb4#diff-d4fe039eee511c339d76db54bac9ed9da96b92c85efbe724ec1f805f2528acc7L1241-R1252

**Review by shawntabrizi — APPROVED** [review]
thanks for getting that figured out

**Review by kianenigma — APPROVED** [review]

**Comment by ggwpez** [comment]
This fixes my original problem. Thanks!

**Review by gavofyork — APPROVED** [review]

**Pull request merged**